File: leetcode/editor/en/[1183]Maximum Number of Ones.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number} width
 * @param {number} height
 * @param {number} sideLength
 * @param {number} maxOnes
 * @return {number}
 */
var maximumNumberOfOnes = function (width, height, sideLength, maxOnes) {
    let arr = new Array(sideLength * sideLength).fill(0);
    for (let i = 0; i < height; i++) {
        let row = i % sideLength;
        for (let j = 0; j < width; j++) {
            let col = j % sideLength;
            arr[(row * sideLength) + col]++;
        }
    }
    arr.sort((a, b) => b - a);
    let res = 0;
    for (let i = 0; i < maxOnes; i++) {
        res += arr[i];
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[1183]Maximum Number of Ones.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int maximumNumberOfOnes(int width, int height, int sideLength, int maxOnes) {
        int n = sideLength * sideLength;
        int[] arr = new int[n];
        for (int i = 0; i < height; i++) {
            int row = i % sideLength;
            for (int j = 0; j < width; j++) {
                int col = j % sideLength;
                arr[(row * sideLength) + col]++;
            }
        }
        Arrays.sort(arr);
        int res = 0;
        for (int i = 0; i < maxOnes; i++) {
            res += arr[n - 1 - i];
        }
        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
